src/Catalyst.Core.Modules.Dfs/CoreApi/BlockApi.cs

```diff
@@ -0,0 +1,396 @@
+using Lib.P2P;
+using MultiFormats;
+
+namespace Catalyst.Core.Modules.Dfs.CoreApi
+{
+    [DataContract]
+    internal class DataBlock : IDataBlock
+    {
+        [DataMember]
+        public byte[] DataBytes { get; set; }
+
+        public Stream DataStream => new MemoryStream(DataBytes, false);
+
+        [DataMember]
+        public Cid Id { get; set; }
+
+        [DataMember]
+        public long Size { get; set; }
+    }
+
+    internal sealed class BlockApi : IBlockApi
+    {
+        private static readonly ILog Log = LogManager.GetLogger(typeof(BlockApi));
+
+        private static DataBlock GetEmptyDirectory()
+        {
+            var emptyDirectory = ObjectApi.GetEmptyDirectory();
+            return new DataBlock
+            {
+                DataBytes = emptyDirectory.ToArray(),
+                Id = emptyDirectory.Id,
+                Size = emptyDirectory.ToArray().Length
+            };
+        }
+
+        private static DataBlock GetEmptyNode()
+        {
+            var emptyNode = ObjectApi.GetEmptyNode();
+            return new DataBlock
+            {
+                DataBytes = emptyNode.ToArray(),
+                Id = emptyNode.Id,
+                Size = emptyNode.ToArray().Length
+            };
+        }
+
+        private FileStore<Cid, DataBlock> _store;
+
+        private readonly IDhtApi _dhtApi;
+        private readonly ISwarmApi _swarmApi;
+        private readonly IBitSwapApi _bitSwapApi;
+        private readonly DfsOptions _dfsOptions;
+        private readonly DfsState _dfsState;
+
+        public IPinApi PinApi { get; set; }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="bitSwapApi"></param>
+        /// <param name="dhtApi"></param>
+        /// <param name="swarmApi"></param>
+        /// <param name="dfsOptions"></param>
+        public BlockApi(IBitSwapApi bitSwapApi,
+            IDhtApi dhtApi,
+            ISwarmApi swarmApi,
+            DfsState dfsState,
+            DfsOptions dfsOptions)
+        {
+            _bitSwapApi = bitSwapApi;
+            _dhtApi = dhtApi;
+            _swarmApi = swarmApi;
+            _dfsOptions = dfsOptions;
+            _dfsState = dfsState;
+        }
+
+        public FileStore<Cid, DataBlock> Store
+        {
+            get
+            {
+                if (_store != null)
+                {
+                    return _store;
+                }
+
+                var folder = Path.Combine(_dfsOptions.Repository.Folder, "blocks");
+
+                if (!Directory.Exists(folder))
+                {
+                    Directory.CreateDirectory(folder);
+                }
+
+                _store = new FileStore<Cid, DataBlock>
+                {
+                    Folder = folder,
+                    NameToKey = cid => cid.Hash.ToBase32(),
+                    KeyToName = key =>
+                        new MultiHash(key.FromBase32()), // @TODO assumes base32 should get from HashLibProvider prop
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)